scipy/stats/mstats_basic.py

```diff
@@ -839,10 +839,11 @@ def ttest_1samp(a, popmean, axis=0):
     x = a.mean(axis=axis)
     v = a.var(axis=axis, ddof=1)
     n = a.count(axis=axis)
-    df = n - 1.
-    svar = ((n - 1) * v) / df
-    t = (x - popmean) / ma.sqrt(svar / n)
-    prob = _betai(0.5*df, 0.5, df/(df + t*t))
+    df = n - 1.0
+    with np.errstate(divide='ignore', invalid='ignore'):
+        svar = ((n - 1.0) * v) / df
+        t = (x - popmean) / ma.sqrt(svar / n)
+        prob = special.betainc(0.5*df, 0.5, df/(df + t*t))
```

Comment:
I tracked down the source of the error a bit further. It boils down to the following throwing an Runtime warning:
np.float64(1.0) / ma.masked_array([0.0], mask=[1.0])
Surprisingly (?) the following does not have the same problem
`print(type(np.array([1.0]) - 1.0))`